kore/src/Kore/Step/Simplification/OrPattern.hs

```diff
@@ -24,10 +39,27 @@ import qualified Kore.Step.SMT.Evaluator as SMT.Evaluator
     )
 
 simplifyPredicatesWithSmt
-    :: (MonadSimplify simplifier, SimplifierVariable variable)
-    => OrPattern variable -> simplifier (OrPattern variable)
-simplifyPredicatesWithSmt unsimplified = do
+    ::  forall variable simplifier
+    .   (MonadSimplify simplifier, SimplifierVariable variable)
+    => Syntax.Predicate variable
+    -> OrPattern variable
+    -> simplifier (OrPattern variable)
+simplifyPredicatesWithSmt predicate' unsimplified = do
     simplifiedOrs <- traverse
-        (BranchT.gather . Pattern.simplifyPredicate)
+        (BranchT.gather . Conditional.simplifyPredicate)
         (MultiOr.extractPatterns unsimplified)
-    SMT.Evaluator.filterMultiOr (MultiOr.make (concat simplifiedOrs))
+    let newOrs :: [Conditional variable (Pattern variable)]
+        newOrs = addPredicate . conditionalAsTerm <$> concat simplifiedOrs
```

Comment:
Could you add a comment here that says what is "new" about the `newOrs`? It looks to me like they have another `Conditional` wrapped around the original `Pattern`, and the outer conditions have the new predicate, is that right?

Reply:
Right. I renamed the variables and added a comment.